invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/uploads.js

```diff
@@ -40,24 +40,34 @@ const statuses = {
   new_version_draft: { color: "neutral", title: i18next.t("New version draft") },
 };
 
+const executeRequest = (url, method = "GET") => {
+  return http({
+    url: url,
+    method: method,
+    headers: {
+      "Content-Type": "application/json",
+      "Accept": "application/vnd.inveniordm.v1+json",
+    },
+  });
+};
+
 export const RDMRecordResultsListItem = ({ result }) => {
   const editRecord = () => {
-    http
-      .post(
-        `/api/records/${result.id}/draft`,
-        {},
-        {
-          headers: {
-            "Content-Type": "application/json",
-            "Accept": "application/vnd.inveniordm.v1+json",
-          },
-        }
-      )
+    executeRequest(`/api/records/${result.id}/draft`, "POST")
       .then(() => {
         window.location = `/uploads/${result.id}`;
       })
       .catch((error) => {
-        console.error(error.response.data);
+        if (error.response.status === 403) {
```

Comment:
Due to lack of knowing on a search result the type of access that a user has , this is a workaround to make "Edit" button redirect to preview if you have been shared with preview access only.